src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsDocumentHelper.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Xml.XPath;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen
+{
+    internal static class XmlCommentsDocumentHelper
+    {
+        internal static IReadOnlyDictionary<string, XPathNavigator> GetMemberDictionary(XPathDocument xmlDoc)
+        {
+            var members = xmlDoc.CreateNavigator()
+                .SelectFirstChild("doc")
+                ?.SelectFirstChild("members")
+                ?.SelectChildren("member", "")
+                ?.OfType<XPathNavigator>();
+
+            if (members == null)
+            {
+                return new Dictionary<string, XPathNavigator>();
+            }
+
+            return members.ToDictionary(memberNode => memberNode.GetAttribute("name", ""));
```

Comment:
Is this guaranteed to never have any key collisions?

Reply:
If they're documented as unique then it should be fine to use `ToDictionary()` rather than manually loop and overwrite (or not) any conflicts.